clang/lib/Driver/ToolChains/Patmos.cpp

```diff
@@ -478,21 +478,13 @@ static std::string get_patmos_gold(const ToolChain &TC, DiagnosticsEngine &Diag)
     }
   }
 
-  auto tmp = TC.GetProgramPath("patmos-ld");
-  if (tmp != "patmos-ld") {
-    return tmp;
-  }else{
-    auto DiagID = Diag.getCustomDiagID(DiagnosticsEngine::Error,
-                         "gold linker 'patmos-ld' not found");
-    Diag.Report(DiagID);
-    return "";
-  }
+  return(get_patmos_tool(TC, "ld.lld"));
```

Comment:
We need LLD to be prefixed by `patmos-`,  so that it doesn't clash with a regular installation of LLVM that a user might have. 
```suggestion
  return(get_patmos_tool(TC, "patmos-ld.lld"));
```